pkg/sensors/tracing/kprobe_test.go

```diff
@@ -2389,6 +2389,75 @@ spec:
 	assert.NoError(t, err)
 }
 
+func TestKprobeTraceCapabilityChecks(t *testing.T) {
+	var doneWG, readyWG sync.WaitGroup
+	defer doneWG.Wait()
+
+	ctx, cancel := context.WithTimeout(context.Background(), tus.Conf().CmdWaitTime)
+	defer cancel()
+
+	pidStr := strconv.Itoa(int(observer.GetMyPid()))
+	t.Logf("tester pid=%s\n", pidStr)
+
+	capabilityhook_ := `
+apiVersion: cilium.io/v1alpha1
+kind: TracingPolicy
+metadata:
+  name: "trace-capability-checks"
+spec:
+  kprobes:
+  - call: "cap_capable"
+    syscall: false
+    return: true
+    args:
+    - index: 0
+      type: "nop"
+    - index: 1
+      type: "user_namespace"
+    - index: 2
+      type: "capability"
+    - index: 3
+      type: "nop"
+    returnArg:
+      type: "int"
+    selectors:
+    - matchPIDs:
+      - operator: In
+        values:
+        - ` + pidStr
+
+	createCrdFile(t, capabilityhook_)
+
+	obs, err := observer.GetDefaultObserverWithFile(t, ctx, testConfigFile, tus.Conf().TetragonLib)
+	if err != nil {
+		t.Fatalf("GetDefaultObserverWithFile error: %s", err)
+	}
+	observer.LoopEvents(ctx, t, &doneWG, &readyWG, obs)
+	readyWG.Wait()
+
+	kpChecker := ec.NewProcessKprobeChecker().
+		WithFunctionName(sm.Full("cap_capable")).
+		WithAction(tetragon.KprobeAction_KPROBE_ACTION_POST)
```

Comment:
It should be good if we can check more here. For example, based on the comment below we expect "capability" to be `CAP_SYS_RAWIO`. Maybe we can also check the user namespace value (we have some helpers in `pkg/reader/namespace/namespace.go`).

Reply:
Fixed! thank you @tpapagian